footerTest.js

```diff
@@ -0,0 +1,70 @@
+const puppeteer = require('puppeteer');
+
+let config = {
+  launchOptions: {
+    headless: false,
+    ignoreHTTPSErrors: true,
+  },
+};
+
+let urls = [
+  'https://dashboard.realdevsquad.com/index.html',
+  'https://dashboard.realdevsquad.com/profile/index.html',
+  'https://dashboard.realdevsquad.com/goal/index.html',
+  'https://dashboard.realdevsquad.com/task/index.html',
+  'https://dashboard.realdevsquad.com/users/index.html',
+  'https://dashboard.realdevsquad.com/users/details/index.html',
+  'https://dashboard.realdevsquad.com/taskevents/index.html',
+  'https://dashboard.realdevsquad.com/featureflag/index.html',
+  'https://dashboard.realdevsquad.com/wallet/index.html',
+  'https://dashboard.realdevsquad.com/online-members/online-members.html',
+];
+
+const test_footer = async (url, index) => {
+  await puppeteer.launch(config.launchOptions).then(async (browser) => {
+    const page = await browser.newPage();
+    await page.goto(url, { waitUntil: ['load', 'networkidle2'] });
+    await page.content();
+    page.setDefaultNavigationTimeout(0);
+    const footer = await page.$('footer');
+    if (footer) {
+      // is the text same in footer?
+      const footer_text = await page.evaluate(() => {
+        const element = document.querySelector('.info-repo');
+        console.log(element);
+
+        return (
+          element &&
+          element.innerText ===
+            'The contents of this website are deployed from this open sourced repo'
+        );
+        // will return undefined if the element is not found
+      });
+      console.log(`✅ footer text:${footer_text}  index:${index}:${url}`);
+
+      if (footer_text) {
+        // links inside footer are working or not
+        await Promise.all([
+          page.click('.info-repo a', { delay: 2000 }),
+          page.waitForNavigation(),
+          setTimeout(() => {
```

Comment:
This is a slightly hacky solution.

Let's say our test runs in 1s. Now, we will have to wait extra 9s for the test to exit. If we are running 100+ tests, then we would have an additional 15min delay on our CI for such wait times.

Additionally, if our test was going to run in 11s, your setTimout closes the browser in 10s and reports the test as failed.(False Negative -- Though this is a lesser problem, since we have test timeouts setup anyway)

Reply:
Let's refactor some of it using the working setup in [this branch](https://github.com/Real-Dev-Squad/website-tests/tree/tmp-ankush-single-test-working)

Reply:
Yup will go through it

Reply:
@ankushdharkar can I use headless chrome? All the problems resolved you mentioned using it.